src/Lean/Meta/LevelDefEq.lean

```diff
@@ -81,6 +81,11 @@ mutual
         match (← Meta.decLevel? v) with
         | some v => Bool.toLBool <$> isLevelDefEqAux u v
         | none   => return LBool.undef
+    -- Handle `max u ?_ =?= max u v` and `max u ?_ =?= max v u` for `u` a parameter.
+    | Level.max p@(.param _) m@(.mvar _), Level.max u v =>
+      if u == p then solve m v
+      else if v == p then solve m u
+      else return LBool.undef
```

Comment:
```suggestion
    | Level.max p@(.param _) (.mvar m), Level.max u v =>
      if u == p then
        assignLevelMVar m v
        return LBool.true
      else if v == p then
        assignLevelMVar m u
        return LBool.true
      else return LBool.undef
```
also works here, and is perhaps more conservative as there's no recursive call to `solve`, but I think should have the same behaviour. It also compiles mathlib4 fine, and fixes our problems there.